src/client/display_configuration.h

```diff
@@ -27,10 +27,33 @@
 #include <vector>
 #include <memory>
 
-struct MirDisplayConfig : mir::protobuf::DisplayConfiguration
+struct MirDisplayConfig
 {
-    explicit MirDisplayConfig(mir::protobuf::DisplayConfiguration& pconfig) :
-        mir::protobuf::DisplayConfiguration::DisplayConfiguration{std::move(pconfig)} {}
+    MirDisplayConfig()
+    {
+    }
+
+    explicit MirDisplayConfig(mir::protobuf::DisplayConfiguration& wrapped) :
+        wrapped{std::move(wrapped)}
+    {
+    }
+
+    operator mir::protobuf::DisplayConfiguration const&() const
+    {
+        return wrapped;
+    }
+
+    mir::protobuf::DisplayConfiguration* operator->()
+    {
+        return &wrapped;
+    }
+
+    mir::protobuf::DisplayConfiguration const* operator->() const
+    {
+        return &wrapped;
+    }
```

Comment:
These operators don't help anything: they are not transparent to code that used a `MirDisplayConfig*`.

I can see from the diff there are only a few call sites, and these only use a few functions. I'd prefer to provide forwarding methods and leave the call sites without the weird `(*config)->foo()` incantation this requires.
```c++
    int ByteSize() const { return wrapped.ByteSize(); }
    uint8_t* SerializeWithCachedSizesToArray(uint8_t* target) const { return wrapped.SerializeWithCachedSizesToArray(target); }
    bool ParseFromArray(const void* data, int size) { return wrapped.ParseFromArray(data, size); }
    int display_output_size() const { return wrapped.display_output_size(); }
    mir::protobuf::DisplayOutput const& display_output(int index) const { return wrapped.display_output(index); }
```